op-supervisor/supervisor/backend/syncnode/reset.go

```diff
@@ -0,0 +1,254 @@
+package syncnode
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+// resetTracker manages a bisection
+// between consistent and inconsistent blocks
+// and is used to prepare a reset request
+// which is sent to the managed node
+type resetTracker struct {
+	a eth.BlockID
+	z eth.BlockID
+
+	synchronous bool
+	resetting   *atomic.Bool
+	cancelling  *atomic.Bool
+
+	managed *ManagedNode
+}
+
+// init initializes the reset tracker with
+// empty start and end of range, and no reset in progress
+func (t *resetTracker) init() {
+	t.resetting.Store(true)
+	t.cancelling.Store(false)
+	t.a = eth.BlockID{}
+	t.z = eth.BlockID{}
+}
+
+// beginBisectionReset initializes the reset tracker
+// and starts the bisection process at the given block
+// which will lead to a reset request
+func (t *resetTracker) beginBisectionReset(z eth.BlockID) {
+	// only one reset can be in progress at a time
+	if t.resetting.Load() {
+		return
```

Comment:
Would it be useful to log that this is happening? A lot of duplicate reset requests seems like it would be a case of either 1) sub-optimal code or 2) an important event in the system we'd want to know about.